src/v3/src/util/buildFieldLevelErrorMessages.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { WidgetMessage } from '../types';
+
+type FieldLevelErrorMessages = {
+  errorMessage?: string;
+  errorMessageList?: string[];
+};
+
+export const buildFieldLevelErrorMessages = (
+  errors: WidgetMessage[] | undefined,
+) : FieldLevelErrorMessages => {
+  const hasErrors = typeof errors !== 'undefined' && errors.length > 0;
+
+  if (hasErrors) {
+    if (errors.length === 1) {
+      const error = errors[0];
+      // Covers nested messages, e.g. password requirements not met
+      if (Array.isArray(error.message)) {
+        return {
+          errorMessage: error.description,
+          errorMessageList: error.message
+            .filter((msg) => typeof msg.message === 'string')
+            .map((msg) => msg.message as string),
+        };
+      }
```

Comment:
Confirmed with @glenfannin-okta  that we only ever see one level of "nested messages" in the SIW, so this utility function does not have to be recursive. The **only** case where we even see nesting within field level error messages is the password requirements not met error with a list of requirements that haven't been met (see screenshot). In this case, we use the `errorMessage` as the place to pass in the `error.description` and the `errorMessageList` to pass in the list of errors

![Screenshot 2024-01-30 at 5 36 53 PM](https://github.com/okta/okta-signin-widget/assets/106110543/13fae7e0-d417-49f9-a5c4-59849b5ad1d3)